src/transactions/dapp.js

```diff
@@ -21,6 +21,20 @@ import constants from '../constants';
 import slots from '../time/slots';
 import { prepareTransaction } from './utils';
 
+const isInt = n => parseInt(n, 10) === n;
+
+const validateOptions = ({ category, name, type }) => {
```

Comment:
Good idea to validate here! Please add the fourth required field, which is `link` as required string input. See more on those fields here:
https://docs.lisk.io/v1.3/docs/lisk-api-080-apps